controller/machine/controller.go

```diff
@@ -135,7 +137,20 @@ func (m *Lifecycle) Remove(obj *v3.Machine) (*v3.Machine, error) {
 	if obj.Status.MachineTemplateSpec == nil {
 		return obj, nil
 	}
-
+	var err error
+	for i := 0; i < 5; i++ {
+		found, err := m.isNodeInAppliedSpec(obj)
+		if err == nil && !found {
+			break
+		}
+		if i == 4 {
+			return obj, fmt.Errorf("Machine [%s] still not deleted from cluster spec", obj.Name)
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
It is not a good idea block the goroutine. Rather just return the error and let the queue backoff.